truenas/zfs-initrd

```diff
@@ -74,29 +74,44 @@ mkdir -p "${1}/sbin"
 cat >"${1}/sbin/start-truenas-install" << EOF
 #!/bin/sh
 
-# Mount CDROM
-mkdir /cdrom
-mount -t iso9660 /dev/disk/by-label/TRUENAS_INSTALL /cdrom
-
-# Update modules
-depmod -a
-
-# Load ZFS
-modprobe zfs
-
-# Load installation packages
-udpkg -i /cdrom/pool/main/t/truenas-installer/*.udeb
-udpkg -i /cdrom/pool/main/p/perl/perl-base*.deb
-udpkg -i /cdrom/pool/main/d/debootstrap/*.udeb
-udpkg -i /cdrom/pool/main/d/debian-archive-keyring/*.udeb
-
-# Start syslogd for tty4
-/sbin/syslogd -m 0 -O /var/log/syslog
-
-# Start the real install on tty0
-cat /etc/inittab | grep -v "^tty0" > /etc/inittab.new
-echo "tty0::respawn:/usr/sbin/truenas-install" >> /etc/inittab.new
-mv /etc/inittab.new /etc/inittab
+for i in 1 2 3 4 5;
+do
+	if [ -e /dev/disk/by-label/TRUENAS_INSTALL ]; then
+		break
+	else
+		echo "Unable to find TRUENAS_INSALL, sleeping for 5 seconds"
+		sleep 5
+	fi
+done
+
+if [ -e /dev/disk/by-label/TRUENAS_INSTALL ]; then
```

Comment:
No need to put this whole section into a big if/else statement. I'd do it this way:

```
if [ ! -e /dev/disk-by-label/TRUENAS_INSTALL ]  ;then
   echo "Unable to find /dev/disk/by-label/TRUENAS_INSTALL, dropping to shell"
   /bin/sh
  exit 1
fi
```

Then do the rest of the normal mount below that. Makes it look cleaner.